flow/Arena.h

```diff
@@ -529,18 +529,19 @@ class StringRef {
 
 	int expectedSize() const { return size(); }
 
-	int compare(StringRef const& other) const {
-		if (std::min(size(), other.size()) > 0) {
-			int c = memcmp(begin(), other.begin(), std::min(size(), other.size()));
-			if (c != 0) return c;
+	comparison compare(StringRef const& other) const {
+		size_t minSize = std::min(size(), other.size());
+		if (minSize != 0) {
+			int c = memcmp(begin(), other.begin(), minSize);
+			if (c != 0) return { c < 0, false, c > 0 };
```

Comment:
Not convinced this is faster than doing `if (c != 0) return c < 0 ? { -1 } : { 1 };`